modules/fbx/fbx_parser/FBXMaterial.cpp

```diff
@@ -329,22 +342,33 @@ Video::Video(uint64_t id, const ElementPtr element, const Document &doc, const s
 					// First time compute size (it could be large like 64Gb and it is good to allocate it once)
 					for (uint32_t tokenIdx = 0; tokenIdx < numTokens; ++tokenIdx) {
 						const Token *dataToken = GetRequiredToken(Content, tokenIdx);
+						IF_FBX_IS_CORRUPT_RETURN;
 						size_t tokenLength = dataToken->end() - dataToken->begin() - 2; // ignore double quotes
 						const char *base64data = dataToken->begin() + 1;
 						const size_t outLength = Util::ComputeDecodedSizeBase64(base64data, tokenLength);
 						if (outLength == 0) {
 							DOMError("Corrupted embedded content found", element);
+							FBX_CORRUPT;
+							return;
 						}
 						targetLength += outLength;
 					}
 					if (targetLength == 0) {
 						DOMError("Corrupted embedded content found", element);
+						FBX_CORRUPT;
+						return;
 					} else {
 						content = new uint8_t[targetLength];
 						contentLength = static_cast<uint64_t>(targetLength);
 						size_t dst_offset = 0;
 						for (uint32_t tokenIdx = 0; tokenIdx < numTokens; ++tokenIdx) {
 							const Token *dataToken = GetRequiredToken(Content, tokenIdx);
+							if (!dataToken) {
+								delete[] content;
```

Comment:
Is there a specific reason why this is done with `new` instead of a `Vector<uint8_t>`?